initfiles/componentfiles/thor/start_slaves

```diff
@@ -35,25 +35,38 @@ fi
 # this must match jsocket hard limit
 export handlelimit=32768
 
-echo `date` Initializing dafilesrv setup
-sudo /etc/init.d/hpcc-init -c dafilesrv setup
-echo `date` Dafilesrv setup done
+
+# initialize logging
+sudo /etc/init.d/dafilesrv setup
 mkdir -p $instancedir
 mkdir -p `dirname $logredirect`
 exec >>$logredirect 2>&1
 
 cd $instancedir
 
-echo "slave(${ip}) init `date`"
+echo "`date` Checking status of dafilesrv dependency"
+sudo /etc/init.d/dafilesrv status &>/dev/null
+if [ $? -ne 0 ];then
+  echo "`date` Attempting to start dafilesrv dependency"
+  sudo /etc/init.d/dafilesrv start &>/dev/null
+  if [ $? -ne 0 ];then
+    echo "`date` FATAL ERROR :: Failed to start dafilesrv dependency"
+    echo "`date` Exiting start_slaves script"
+    exit 1
+  fi
+fi
+echo "`date` Dependency dafilesrv is running"
+
+echo "`date` slave(${ip}) init"
 
 lckfile="$PID/start_slaves_${hpcc_compname}_${ip}.pid"
 
 # prevent two of these scripts starting together
 while [ -e $lckfile ]; do
-  echo waiting on lckfile: $lckfile
+  echo "`date` waiting on lckfile: ${lckfile}"
```

Comment:
Adding the date to this script logging is definitely worth it.
Not for this commit, but I suspect really we should introduce a common function that all scripts use, e.g. log(<msg>), that all HPCC scripts use and standardizes the output format.